storybook/qmlTests/tests/tst_SwapInputPanel.qml

```diff
@@ -0,0 +1,218 @@
+
+            amountToSendInput.input.locale = Qt.locale("cs_CZ")
+            compare(amountToSendInput.input.locale.name, "cs_CZ")
+
+            // manually entering "1,42" meaning "1.42"; `,` being the decimal separator
+            keyClick(Qt.Key_1)
+            keyClick(Qt.Key_Comma)
+            keyClick(Qt.Key_4)
+            keyClick(Qt.Key_2)
+
+            tryCompare(controlUnderTest, "cryptoValue", 1.42)
+            verify(controlUnderTest.cryptoValueValid)
+        }
+
+        function test_selectSTTHoldingAndTypeAmount() {
+            controlUnderTest = createTemporaryObject(componentUnderTest, root)
+            verify(!!controlUnderTest)
+
+            const holdingSelector = findChild(controlUnderTest, "holdingSelector")
+            verify(!!holdingSelector)
+            mouseClick(holdingSelector)
+            waitForRendering(holdingSelector)
+
+            const assetSelectorList = findChild(holdingSelector, "assetSelectorList")
+            verify(!!assetSelectorList)
+            waitForRendering(assetSelectorList)
+
+            const sttDelegate = findChild(assetSelectorList, "AssetSelector_ItemDelegate_STT")
+            verify(!!sttDelegate)
+            mouseClick(sttDelegate, 40, 40) // center might be covered by tags
+
+            tryCompare(controlUnderTest, "selectedHoldingId", "STT")
+
+            const amountToSendInput = findChild(controlUnderTest, "amountToSendInput")
+            verify(!!amountToSendInput)
+            mouseClick(amountToSendInput)
+            waitForRendering(amountToSendInput)
+            verify(amountToSendInput.input.input.edit.activeFocus)
+
+            keyClick(Qt.Key_1)
+            keyClick(Qt.Key_Period)
+            keyClick(Qt.Key_4)
+            keyClick(Qt.Key_2)
+
+            tryCompare(controlUnderTest, "cryptoValue", 1.42)
+            verify(controlUnderTest.cryptoValueValid)
+        }
+
+        function test_clickingMaxButton() {
+            controlUnderTest = createTemporaryObject(componentUnderTest, root, {inputTokenKey: "ETH"})
+            verify(!!controlUnderTest)
+            waitForRendering(controlUnderTest)
+            tryCompare(controlUnderTest, "selectedHoldingId", "ETH")
+
+            const maxTagButton = findChild(controlUnderTest, "maxTagButton")
+            verify(!!maxTagButton)
+            waitForRendering(maxTagButton)
+            verify(maxTagButton.visible)
+            mouseClick(maxTagButton)
+
+            const maxValue = maxTagButton.title.match(/Max: (\d+\.?\d*) ETH/)[1] // extract the value from the max button
+
+            const amountToSendInput = findChild(controlUnderTest, "amountToSendInput")
+            verify(!!amountToSendInput)
+            waitForRendering(amountToSendInput)
+
+            tryCompare(amountToSendInput.input, "text", maxValue)
+            tryCompare(controlUnderTest, "cryptoValue", Number.fromLocaleString(Qt.locale(), maxValue))
+            verify(controlUnderTest.cryptoValueValid)
+        }
```

Comment:
Missing tests for large value check behavior?

Reply:
Test added/adjusted